File: csharp/Driver/Driver.cs
using System.Collections.Generic;

namespace NP6Api
{
    /// <summary>
    /// Driver class that contain everything
    /// </summary>
    public class Driver
    {
        private string _url;
        private string _xKey;
        private Connection _connection;

        public Target Target;
        public Action Action;
        public Field Field;
        public Send Send;
        public Import Import;
        public Segment Segment;

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="config">Configuration for the driver</param>
        public Driver(Dictionary<string, string> config)
        {
            _url = config["url"] != null ? config["url"] : "https://backoffice.mailperformance.com/";
            _xKey = config["xKey"];
            _connection = new Connection(_xKey, _url);
            Target = new Target(_connection);
            Action = new Action(_connection);
            Field = new Field(_connection);
            Send = new Send(_connection);
            Import = new Import(_connection);
            Segment = new Segment(_connection);
        }

    }
}

File: php/src/Driver.php
<?php

spl_autoload_register(function ($class_name) {
    include __DIR__ . '/' . $class_name . '.php';
});

/**
 * Driver class that contain everything.
 */
class Driver
{
    private $_xKey;
    private $_url;
    private $_connection;
    public $segment;
    public $target;
    public $action;
    public $send;
    public $import;
    /**
     * @param string $xKey
     */
    public function __construct($config)
    {
        $this->_xKey = $config['xKey'];
        $this->_url = (isset($config['url'])) ? $config['url'] : 'https://backoffice.mailperformance.com/';
        $this->_connection = new Connection($this->_xKey, $this->_url);
        $this->segment = new Segment($this->_connection);
        $this->target = new Target($this->_connection);
        $this->action = new Action($this->_connection);
        $this->send = new Send($this->_connection);
        $this->import = new Import($this->_connection);
        $this->field = new Field($this->_connection);
    }
    /**
     * @return string
     */
    public function getUrl()
    {
        return $this->_url;
    }
    /**
     * @return string
     */
    public function getXKey()
    {
        return $this->_xKey;
    }
}

?>
